File: leet_804_unique_morese_code_words.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words)
    {
        set<string> E;
        int i = -1;
        while (++i < words.size())  E.insert(mc(words[i]));
        return E.size() ;
    }
    
    string  mc(string s)
    {
        vector<string>  morse = {".-","-...","-.-.","-..",".","..-.",\
            "--.","....","..",".---","-.-",".-..","--","-.","---",".--.",\
            "--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        string  res = "";
        int     i = -1;

        while (++i < s.length())
            res += morse[ s[i] - 'a' ];
        return res ;
    }
};

File: leet_804_unique_morese_code_words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        E = set()
        for s in words:
            E.add( self.mc(s) )
        return len(E)

    def mc(self, s: str) -> str:
        morse = [".-","-...","-.-.","-..",".","..-.","--.",\
                 "....","..",".---","-.-",".-..","--","-.",\
                 "---",".--.","--.-",".-.","...","-","..-",\
                 "...-",".--","-..-","-.--","--.."]
        res = ""
        for c in s:
            res += morse[ord(c) - ord('a')]
        return res
